src/xtgeo/grid3d/grid.py

```diff
@@ -141,100 +257,170 @@ class Grid(_Grid3D):
+        actnumsv: np.ndarray,
+        dualporo: bool = False,
+        dualperm: bool = False,
+        subgrids: OrderedDict = None,
         units: Optional[Units] = None,
-        **kwargs,
+        filesrc: Optional[Union[Path, str]] = None,
+        props: Optional[GridProperties] = None,
+        name: Optional[str] = None,
+        roxgrid=None,
+        roxindexer=None,
     ):
-        """Instantating.
 
-        Args:
-            gfile: Input file, or leave blank.
-            fformat: File format input, default is ``guess`` based on file extension.
-                Other options are ...
-            units: The length units the coordinates are in,
-                (either Units.CM, Units.METRES, Units.FEET for cm, metres and
-                feet respectively).  Default (None) is unitless.
-            kwargs: remaining keyword arguments are sent to the corresponding file
-                reader, see :meth:`Grid.from_file`.
+        coordsv = np.asarray(coordsv)
+        zcornsv = np.asarray(zcornsv)
+        actnumsv = np.asarray(actnumsv)
+        if coordsv.dtype != np.float64:
+            raise TypeError(
+                f"The dtype of the coordsv array must be float64, got {coordsv.dtype}"
+            )
+        if zcornsv.dtype != np.float32:
+            raise TypeError(
+                f"The dtype of the zcornsv array must be float32, got {zcornsv.dtype}"
+            )
+        if actnumsv.dtype != np.int32:
+            raise TypeError(
+                f"The dtype of the actnumsv array must be int32, got {actnumsv.dtype}"
+            )
+        if len(coordsv.shape) != 3 or coordsv.shape[2] != 6:
+            raise ValueError(
+                f"shape of coordsv should be (nx+1,ny+1,6), got {coordsv.shape}"
+            )
+        if len(zcornsv.shape) != 4 or zcornsv.shape[3] != 4:
+            raise ValueError(
+                f"shape of zcornsv should be (nx+1,ny+1,nz+1, 4), got {zcornsv.shape}"
+            )
+        if zcornsv.shape[0:2] != coordsv.shape[0:2]:
+            raise ValueError(
+                f"Mismatch between zcornsv and coordsv shape: {zcornsv.shape}"
+                f" vs {coordsv.shape}"
+            )
+        if np.any(np.asarray(zcornsv.shape[0:3]) != np.asarray(actnumsv.shape) + 1):
+            raise ValueError(
+                f"Mismatch between zcornsv and actnumsv shape: {zcornsv.shape}"
+                f" vs {actnumsv.shape}"
+            )
 
-        Example::
+        super().__init__(*actnumsv.shape)
+
+        self._reset(
+            coordsv=coordsv,
+            zcornsv=zcornsv,
+            actnumsv=actnumsv,
+            xtgformat=2,
+            dualporo=dualporo,
+            dualperm=dualperm,
+            subgrids=subgrids,
+            units=units,
+            filesrc=filesrc,
+            props=props,
+            name=name,
+            roxgrid=roxgrid,
+            roxindexer=roxindexer,
+        )
 
-            import xtgeo
+    def _reset(
```

Comment:
Add a quick comment that this is to be deleted when the deprecated functions are removed?

At least we're starting to gather quite a few number of functions that we will remove eventually, and it could be easy to miss some of them..